README.md

```diff
@@ -92,3 +92,39 @@ explicitly or implicitly down-cased.
 
 For example, the Http header `X-B3-ParentSpanId: 0020000000000001` would become
 an ASCII header `x-b3-parentspanid` with the same value.
+
+# Frequently Asked Questions
+
+## Why is ParentSpanId propagated?
+
+In B3, the trace context is extracted from headers sent upstream. Timing and
+metadata for the client and server side of an operation are recorded against
+the same context. So, the `ParentSpanId` is the ID of the operation that caused
+the current RPC. For example, it could be the ID of another server request
+or a scheduled job. `ParentSpanId` is propagated so that when data is reported
+to Zipkin, it can be placed in the correct spot in the trace tree.
+
+Some propagation formats look similar to B3, but don't propagate a field named
+parent. People with these formats may wonder why B3 has `X-B3-ParentSpanId` and
+they don't. The usual case is that they propagate a span ID field, but it serves
+the same purpose as `ParentSpanId`. Unlike B3, these systems use a different
+span ID for the client and server side of an RPC. When a server reads headers
+like this, it is expected to provision a new span ID for itself, and use the one
+it extracted as its parent. The end result is still three IDs
+(trace, parent, span), allowing data to be reported properly.
+
+Here's an example of a propagation system which sends the caller's ID, and
+expects the server to provision a new span ID:
+
+```
+                              ┌───────────────────┐
+ Incoming Request Headers     │   TraceContext    │
+┌───────────────────┐         │ ┌───────────────┐ │
+│ XXXX─TraceId      │─────────┼─┼> TraceId      │ │
+│                   │extract()│ │               │ │
+│ XXXX─SpanId       │─────────┼─┼> ParentSpanId │ │
+└───────────────────┘         │ │               │ │      ┌──────────────┐
+                              │ │  SpanId      <┼─next()─│ ID Generator │
```

Comment:
First box is a service (probably) , second box is a service but third box is functionality inside a library. That makes it confusing